heron/healthmgr/src/java/com/twitter/heron/healthmgr/resolvers/RestartContainerResolver.java

```diff
@@ -0,0 +1,110 @@
+import com.microsoft.dhalion.diagnoser.Diagnosis;
+import com.microsoft.dhalion.events.EventManager;
+import com.microsoft.dhalion.resolver.Action;
+
+import com.twitter.heron.healthmgr.common.HealthManagerEvents.ContainerRestart;
+import com.twitter.heron.healthmgr.common.PhysicalPlanProvider;
+
+import static com.twitter.heron.healthmgr.HealthManager.CONF_TOPOLOGY_NAME;
+import static com.twitter.heron.healthmgr.diagnosers.BaseDiagnoser.DiagnosisName.DIAGNOSIS_SLOW_INSTANCE;
+
+public class RestartContainerResolver implements IResolver {
+  private static final Logger LOG = Logger.getLogger(RestartContainerResolver.class.getName());
+
+  private PhysicalPlanProvider physicalPlanProvider;
+  private EventManager eventManager;
+  private String topologyName;
+
+  @Inject
+  public RestartContainerResolver(@Named(CONF_TOPOLOGY_NAME) String topologyName,
+      PhysicalPlanProvider physicalPlanProvider, EventManager eventManager) {
+    this.topologyName = topologyName;
+    this.physicalPlanProvider = physicalPlanProvider;
+    this.eventManager = eventManager;
+  }
+
+  @Override
+  public List<Action> resolve(List<Diagnosis> diagnosis) {
+    for (Diagnosis diagnoses : diagnosis) {
+      Symptom bpSymptom = diagnoses.getSymptoms().get(DIAGNOSIS_SLOW_INSTANCE.text());
+      if (bpSymptom == null || bpSymptom.getComponents().isEmpty()) {
+        // nothing to fix as there is no back pressure
+        continue;
+      }
+
+      if (bpSymptom.getComponents().size() > 1) {
+        throw new UnsupportedOperationException("Multiple components with back pressure symptom");
+      }
+
+      try {
+        // TODO: want to know which stmgr has backpressure
+        String stmgrId = bpSymptom.getComponent().getName();
+        String shellUrl = physicalPlanProvider.getShellUrl(stmgrId);
+        if (shellUrl == null) {
+          throw new MalformedURLException("stmgr not found " + stmgrId);
+        }
+        String urlStr = "http://" + shellUrl + "/killexecutor";
+        URL url = new URL(urlStr);
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod("POST");
+
+        con.setDoOutput(true);
+        DataOutputStream out = new DataOutputStream(con.getOutputStream());
+        out.writeBytes("secret=" + topologyName);
+        out.flush();
+        out.close();
+
+        int status = con.getResponseCode();
+        LOG.info("Restarting container: " + url.toString() + "; result: " + status);
+        con.disconnect();
+
+        ContainerRestart action = new ContainerRestart();
+        LOG.info("Broadcasting container restart event");
+        eventManager.onEvent(action);
+
+        List<Action> actions = new ArrayList<>();
+        actions.add(action);
+        return actions;
+      } catch (MalformedURLException e) {
+        LOG.warning(e.getMessage());
+      } catch (IOException e) {
+        LOG.warning(e.getMessage());
+      }
+    }
+
+    return null;
```

Comment:
need @ashvina confirm the expected return value for resolve()
what is difference between empty list and null?